README.md

```diff
@@ -0,0 +1,112 @@
+So in our example the Instance would Assume into the Role having access to the S3 bucket.
+This process alone does not really change the authorization problem as still every application would be able to use it but AWS gives an additional layer of security to this procedure: external_id. The external_id is a pre-shared-key added to the trust relationship of a role which allows us to assume into it.
+This PSK will allow us to use DC/OS Secrets acting as authorization instance for our application by placing an AssumeRole configuration including the external_id.
+
+## Example
+[This repository](https://github.com/fatz/dcos-secure-instance-profiles) contains a [main.tf](./main.tf) with a example setup. You only need to place a DC/OS license in your Homefolder ( `$HOME/license.txt` ) and the public key file of the SSH-Key you've loaded into your ssh-agent at `~/.ssh/id_rsa.pub`. If these files are at different locations just edit the `main.tf` and change the path for your environment.
+
+### Creating the cluster
+Once you've downloaded all the files of this repository ( `git clone https://github.com/fatz/dcos-secure-instance-profiles && cd dcos-secure-instance-profiles` ) you need to initialize terraform and start creating the cluster
+
+Before you start creating the cluster make sure your AWS setup is finished and working. Either `$AWS_PROFILE` needs to be set to the profile you want to use or make sure you've properly setup your aws cli `aws configure`. To ensure you will use the expected account you can run `aws sts get-caller-identity` and see the account id you will be using.
+
+```bash
+terraform init -upgrade .
+terraform apply
+```
+
+### If not already done download the dcos-cli
+
+```bash
+# on OSX
+brew install dcos-cli
+
+# on linux
+
+```
+
+### Attach to cluster
+After successfully creating the cluster we have to attach to the cluster
+
+```bash
+# in this setup we have to use --insecure as we did not give the load balancer a ACM cert and so it is an self signed one.
+dcos cluster setup $(terraform output masters_dns_name) --password=deleteme --username=bootstrapuser --insecure
+```
+
+### Ensure enterprise CLI
+Lets make sure we have the enterprise features available in our CLI ( this is usually just needed for older versions of DC/OS (cli))
+
+```bash
+dcos package install dcos-enterprise-cli --cli --yes
+```
+
+### AWS config secret
+We already prepared the aws config for the application in our `main.tf`. Next we create the secret from it.
+
+```bash
+dcos security secrets create /instance-profile-app/aws-config -v "$(terraform output secret_aws_conf)"
+```
+
+### Install marathon-lb
+To access our app lets install marathon-lb. As we’re running strict mode we’ve to create a service-account and a service-account-secret
```

Comment:
To access our app let's install marathon-lb. As we’re running strict mode we have to create a service-account and a service-account-secret